zerver/lib/actions.py

```diff
@@ -491,7 +491,8 @@ def do_create_user(email: str, password: Optional[str], realm: Realm, full_name:
                    short_name: str, bot_type: Optional[int]=None,
                    is_realm_admin: bool=False, is_guest: bool=False,
                    bot_owner: Optional[UserProfile]=None, tos_version: Optional[str]=None,
-                   timezone: str="", avatar_source: str=UserProfile.AVATAR_FROM_GRAVATAR,
+                   timezone: str="", twenty_four_hour_time: bool=True,
```

Comment:
I think we should keep the `twenty_four_hour_time` as and optional bool and set it as `realm.default_twenty_four_hour_time` when not provided.

The same thing for other methods as well.

Reply:
do I need to do like this?
```suggestion
           def do_create_user(timezone: str="", twenty_four_hour_time: Optional[bool]=None) -> UserProfile:

    if not twenty_four_hour_time:
        twenty_four_hour_time = realm.default_twenty_four_hour_time
```
But i already  set ` twenty_four_hour_time = realm.default_twenty_four_hour_time` in registreation.py file so it can't be None . do i need to add this again in `action.py` file ?
```suggestion
        twenty_four_hour_time = realm.default_twenty_four_hour_time
                   if 'time_format' in request.POST:
        if request.POST['time_format'] == "12":
                twenty_four_hour_time = False
       elif request.POST['time_format'] == "24":
                twenty_four_hour_time = True
```

Reply:
IMO it's just a better way to write the code. Currently, there's no scope of error but if the method is re-used somewhere else, small things like this are usually missed.